src/kinetics/LinearBurkeRate.cpp

```diff
@@ -0,0 +1,351 @@
+//! @file LinearBurkeRate.cpp
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#include "cantera/kinetics/LinearBurkeRate.h"
+#include "cantera/thermo/ThermoPhase.h"
+#include "cantera/kinetics/Kinetics.h"
+#include <boost/variant.hpp>
+
+namespace Cantera
+{
+
+LinearBurkeData::LinearBurkeData()
+{
+    moleFractions.resize(1, NAN);
+}
+
+bool LinearBurkeData::update(const ThermoPhase& phase, const Kinetics& kin)
+{
+    int X = phase.stateMFNumber();
+    double T = phase.temperature();
+    double P = phase.pressure();
+    if (moleFractions.empty()) {
+        moleFractions.resize(kin.nTotalSpecies());
+    }
+    if (P != pressure || T != temperature || X != mf_number) {
+        ReactionData::update(T);
+        pressure = P;
+        logP = std::log(P);
+        mf_number = X;
+        phase.getMoleFractions(moleFractions.data());
+        return true;
+    }
+    return false;
+}
+
+void LinearBurkeData::perturbPressure(double deltaP)
+{
+    if (m_pressure_buf > 0.) {
+        throw CanteraError("LinearBurkeData::perturbPressure",
+            "Cannot apply another perturbation as state is already perturbed.");
+    }
+    m_pressure_buf = pressure;
+    update(temperature, pressure * (1. + deltaP));
+}
+
+void LinearBurkeData::restore()
+{
+    ReactionData::restore();
+    if (m_pressure_buf < 0.) {
+        return;
+    }
+    update(temperature, m_pressure_buf);
+    m_pressure_buf = -1.;
+}
+
+LinearBurkeRate::LinearBurkeRate(const AnyMap& node, const UnitStack& rate_units)
+{
+    setParameters(node, rate_units);
+}
+
+void LinearBurkeRate::setParameters(const AnyMap& node, const UnitStack& rate_units)
+{
+    UnitStack eps_units{{Units(1.0), 1.0}};
+    ReactionRate::setParameters(node, rate_units);
+    auto eqn = node["equation"].asString();
+    if (!node.hasKey("colliders")) {
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
+            "'colliders' key missing from reaction '{}'.",eqn);
+    }
+    auto colliders = node["colliders"].asVector<AnyMap>();
+    if (!colliders[0].hasKey("name")) {
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
+            "'colliders' key missing from reaction '{}'.",eqn);
+    }
+    else if (!colliders[0].hasKey("eps") && !colliders[0].hasKey("eig0")) {
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
```

Comment:
The formatting for `else if` used throughout Cantera is:
```suggestion
            "'colliders' key missing from reaction '{}'.",eqn);
    } else if (!colliders[0].hasKey("eps") && !colliders[0].hasKey("eig0")) {
        throw InputFileError("LinearBurkeRate::setParameters", m_input,
```